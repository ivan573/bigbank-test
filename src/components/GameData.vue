<template>
  <div class="game-data">
      <span class="game-data__item">Lives: {{ data.lives }}</span>
      <span class="game-data__item">Gold: {{ data.gold }}</span>
      <span class="game-data__item">Level: {{ data.level }}</span>
      <span class="game-data__item">Turn: {{ data.turn }}</span>
      <span class="game-data__item">Score: {{ data.score }}</span>
    </div>
</template>

<script>
export default {
  name: 'GameData',
  props: {
    data: Object
  }
}
</script>

<style lang="scss" scoped>
  .game-data {
    display: flex;
    justify-content: center;
    border-top: $border;
    border-bottom: $border;

    &__item {
      padding: 4px 8px;
      border-left: $border;

      &:last-child {
        border-right: $border;
      }
    }
  }
</style>
