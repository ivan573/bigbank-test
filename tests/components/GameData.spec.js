import { shallowMount } from '@vue/test-utils'
import GameData from '@/components/GameData.vue'

describe('GameData.vue', () => {
  const data = {
    lives: 3,
    gold: 0,
    level: 0,
    turn: 0,
    score: 0
  }

  const wrapper = shallowMount(GameData, {
    propsData: { data }
  })

  it('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('displays all the data from props properly', () => {
    const itemElements = wrapper.findAll('.game-data__item')

    expect(itemElements.at(0).text()).toMatch(`Lives: ${data.lives}`)
    expect(itemElements.at(1).text()).toMatch(`Gold: ${data.gold}`)
    expect(itemElements.at(2).text()).toMatch(`Level: ${data.level}`)
    expect(itemElements.at(3).text()).toMatch(`Turn: ${data.turn}`)
    expect(itemElements.at(4).text()).toMatch(`Score: ${data.score}`)
  })
})
