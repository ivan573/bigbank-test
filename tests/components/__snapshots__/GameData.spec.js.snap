// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameData.vue matches snapshot 1`] = `
<div
  class="game-data"
>
  <span
    class="game-data__item"
  >
    Lives: 3
  </span>
   
  <span
    class="game-data__item"
  >
    Gold: 0
  </span>
   
  <span
    class="game-data__item"
  >
    Level: 0
  </span>
   
  <span
    class="game-data__item"
  >
    Turn: 0
  </span>
   
  <span
    class="game-data__item"
  >
    Score: 0
  </span>
</div>
`;
